apps/rush-lib/src/api/RushConfiguration.ts

```diff
@@ -363,6 +393,34 @@ export class PnpmOptionsConfiguration extends PackageManagerOptionsConfiguration
    */
   public readonly useWorkspaces: boolean;
 
+  /**
+   * See https://pnpm.io/package_json#pnpmneverbuiltdependencies
+   *
+   * @remarks
+   * This field allows to ignore the builds of specific dependencies. The "preinstall", "install", and
+   * "postinstall" scripts of the listed packages will not be executed during installation.
+   */
+  public readonly neverBuiltDependencies: string[] | undefined;
+
+  /**
+   * See https://pnpm.io/package_json#pnpmpackageextensions
+   *
+   * @remarks
+   * The packageExtensions fields offer a way to extend the existing package definitions with additional
+   * information.
+   */
+  public readonly packageExtensions: IPnpmPackageExtensions | undefined;
+
+  /**
+   * See https://pnpm.io/package_json#pnpmoverrides
+   *
+   * @remarks
+   * This field allows you to instruct pnpm to override any dependency in the dependency graph. This is
+   * useful to enforce all your packages to use a single version of a dependency, backport a fix, or
+   * replace a dependency with a fork.
+   */
+  public readonly overrides: Record<string, string> | undefined;
```

Comment:
Same question - how does this interact with `.pnpmfile.cjs`? 

It seems fine to use either `rush.js` or `.pnpmfile.cjs` to adjust dependencies, but of BOTH are being used at the same time, it seem potentially confusing for developers. They might not be aware that the other file is influencing things, and even if they are, they might have trouble understanding how the two representations interact.

Whereas I could see value in a model where:
- We recommend to try to represent your fixes using these settings instead of `.pnpmfile.cjs` (because data is better than script code)
- But if your mapping problem is too complicated, then `.pnpmfile.cjs` is available as an option
- But Rush will not allow you to use both simultaneously

Reply:
I agree with the first two points, but I don't think blocking using both `.pnpmfile.cjs` and this is useful. PNPM knows how to handle that, so I don't think we should restrict that. We should just recommend using this first.

This honestly seems like it'd fit better with the options described in `common-versions.json`. What about renaming that file to `package-resolution-options.json` (or something along those lines) and sticking these options there?

Reply:
To make this more concrete, suppose in one file somebody wrote:

**rush.json**
```js
    "overrides": {
      "qar@1>zoo": "2"
    }
```

...and in another file somebody did:

**.pnpmfile.cjs**
```js
function readPackage(packageJson, context) {
  if (packageJson.name === 'qar') {
    packageJson.dependencies["zoo"] = "1.2.1"
  }
  return packageJson;
}
```

1. Which one takes precedence?  The PNPM docs don't say
2. The filters `qar@1>zoo` and `packageJson.name === 'qar'` are not equivalent, so there are edge cases where one or the other override may apply.  Isn't that a little confusing?
3. If somebody is fiddling with **.pnpmfile.cjs** and was not aware of the other setting in **rush.json**, then **.pnpmfile.cjs** seems to work except in those edge cases. That would be very confusing. They may misinterpret the problem, e.g. assume that Rush or PNPM is broken somehow. How can we prevent that?

Reply:
Does PNPM have an option to verbosely log how it does package resolution and/or dump whatever it's thinking about during installation? Something similar to TypeScript's `traceResolution` option? We could expose a way to write that to a file in `--debug` mode or something and recommend looking through that if someone isn't sure what's going on.

Reply:
@octogonz 

> We recommend to try to represent your fixes using these settings instead of .pnpmfile.cjs (because data is better than script code)
> But if your mapping problem is too complicated, then .pnpmfile.cjs is available as an option

Yep. The idea of this PR is to support the functionality from pnpm as much as possible.

I would like to say pnpm provides `.pnpmfile.cjs` is a nuclear weapon. You can fix almost everything in `readPackage`, but as you said, it's script code. while `pnpm.xxx` configurations are data, which is better and recommended by rush.

Therefore, i think it's OK to use both of them, and `pnpmOptions` in rush.json would be recommended officially.

Reply:
@iclanton 

> Does PNPM have an option to verbosely log how it does package resolution and/or dump whatever it's thinking about during installation?

For now, when you specify `rush install --debug-package-manager` will internally specifies `pnpm install --report ndjson`. It logs sth like

```
{"time":1649349616101,"hostname":"<your_host_name>","pid":17402,"level":"debug","name":"pnpm:progress","packageId":"registry.npmjs.org/sourcemap-codec/1.4.8","requester":"/Users/admin/workspace/rollup-plugin-monaco-editor","status":"resolved"}
```